            HackBio

**Track 3: Data Science & Machine Learning**

(**Topic:** **5 Bioinformatics Databases of Cancer**)

 

A database is considered cancer-related if its content is primarily focused on the topic of cancer. These databases often include cancer-specific data, such as genetic data, molecular profiles, clinical information, and other datasets. As an example,

[**The Cancer Genome Atlas (TCGA)**](https://www.cancer.gov/ccg/research/genome-sequencing/tcga) is a well-known cancer-related database that discusses genetic and molecular events in detail. The National Institutes of Health (NIH) founded TCGA to uncover major genomic changes, and it now contains over 11,000 cases across 33 tumor types, offering a massive dataset of molecular alterations. This effort brought together researchers from many domains. approximately a 12-year period, TCGA generated almost 2.5 petabytes of multi-omics data, dramatically improving cancer diagnosis, treatment, and prevention. The research community continues to have access to this vital data. In one study, The TCGA categorized endometrial cancers into four molecular subgroups based on clinical outcomes and treatment responses.

<!--[if gte vml 1]><v:shapetype
 id="_x0000_t75" coordsize="21600,21600" o:spt="75" o:preferrelative="t"
 path="m@4@5l@4@11@9@11@9@5xe" filled="f" stroked="f">
 <v:stroke joinstyle="miter"/>
 <v:formulas>
  <v:f eqn="if lineDrawn pixelLineWidth 0"/>
  <v:f eqn="sum @0 1 0"/>
  <v:f eqn="sum 0 0 @1"/>
  <v:f eqn="prod @2 1 2"/>
  <v:f eqn="prod @3 21600 pixelWidth"/>
  <v:f eqn="prod @3 21600 pixelHeight"/>
  <v:f eqn="sum @0 0 1"/>
  <v:f eqn="prod @6 1 2"/>
  <v:f eqn="prod @7 21600 pixelWidth"/>
  <v:f eqn="sum @8 21600 0"/>
  <v:f eqn="prod @7 21600 pixelHeight"/>
  <v:f eqn="sum @10 21600 0"/>
 </v:formulas>
 <v:path o:extrusionok="f" gradientshapeok="t" o:connecttype="rect"/>
 <o:lock v:ext="edit" aspectratio="t"/>
</v:shapetype><v:shape id="Picture_x0020_4" o:spid="_x0000_s1027" type="#_x0000_t75"
 alt="A screenshot of a computer&#10;&#10;Description automatically generated"
 style='position:absolute;left:0;text-align:left;margin-left:4.35pt;
 margin-top:16.45pt;width:451.3pt;height:205.9pt;z-index:251659264;
 visibility:visible;mso-wrap-style:square;mso-wrap-distance-left:9pt;
 mso-wrap-distance-top:0;mso-wrap-distance-right:9pt;
 mso-wrap-distance-bottom:0;mso-position-horizontal:absolute;
 mso-position-horizontal-relative:text;mso-position-vertical:absolute;
 mso-position-vertical-relative:text'>
 <v:imagedata src="file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_image001.png"
  o:title="A screenshot of a computer&#10;&#10;Description automatically generated"/>
 <w:wrap type="square"/>
</v:shape><![endif]--><!--[if !vml]-->![A screenshot of a computer

Description automatically generated](file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_image002.png)<!--[endif]-->

The [**cBioPortal**](https://www.cbioportal.org/), established by Memorial Sloan Kettering Cancer Center was specifically designed to lower the barriers of access to the complex data sets for Cancer Genomics  . The Centre for Molecular Oncology at MSK hosts the public cBioPortal website. The software is now developed and maintained by a multi-institutional team, consisting of MSK, the Dana Farber Cancer Institute, Princess Margaret Cancer Centre in Toronto, Children's Hospital of Philadelphia, Caris Life Sciences, The Hyve and SE4BIO in the Netherlands, and Bilkent University in Ankara, Turkey. Non-synonymous mutations, DNA copy-number data (putative, discrete values per gene, e.g. "deeply deleted" or "amplified", as well as log2 or linear copy number data), mRNA and microRNA expression data, protein-level and phosphoprotein-level data (mass spectrometry-based), DNA methylation data, and de-identified clinical data are all supported and stored by the portal. The website offers graphical summaries of gene-level data from several platforms, network visualization and analysis, survival analysis, patient-specific inquiries, and software programmatic access. The portal's user-friendly Web interface allows researchers and physicians to access complicated cancer genomes profiles without requiring bioinformatics skills, thus accelerating biological discoveries.

 

 <!--[if gte vml 1]><v:shape
 id="Picture_x0020_2" o:spid="_x0000_i1027" type="#_x0000_t75" style='width:451pt;
 height:222.5pt;visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_image003.png"
  o:title=""/>
</v:shape><![endif]--><!--[if !vml]-->![](file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_image004.png)<!--[endif]-->

The Human Genome Browser at [**UCSC** **(University of California Santa Cruz)** ](https://genome.ucsc.edu/) acts as a portal for displaying various genomic features, including gene predictions, alignments, polymorphisms, and more. The website provides a user-friendly visual interface for genomic data, aligning annotation tracks with genome coordinates to allow for quick and easy correlation of different data types. Many tracks are contributed by global scientists, while others are created by the UCSC Genome Bioinformatics group utilizing public sequences. Users can also upload and visualize their own data using custom tracks.

<!--[if gte vml 1]><v:shape id="Picture_x0020_5" o:spid="_x0000_i1026"
 type="#_x0000_t75" style='width:451pt;height:206pt;visibility:visible;
 mso-wrap-style:square'>
 <v:imagedata src="file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_image005.png"
  o:title=""/>
</v:shape><![endif]--><!--[if !vml]-->![](file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_image006.png)<!--[endif]-->

 
The [**National Cancer Institute Genomic Data Commons (GDC)**](https://gdc.cancer.gov/) is another prominent cancer database that stores, analyses, and shares genetic and clinical data from cancer patients. The GDC seeks to generalize access to cancer genomic data and encourage data sharing among researchers. The GDC helps advance cancer diagnosis and treatment by making precision medicine approaches easier to apply. OpenGDC, derived from the GDC, builds on the existing platform by including the Genomic Data Model. It adds new genomic data in Browser Extensible Data (BED) format and offers associated information in a table-limited key-value format. It improves the efficiency of retrieving genomic and clinical data while also increasing the quantity of information available for analysis.&#x20;

 

<!--[if gte vml 1]><v:shape id="Picture_x0020_1" o:spid="_x0000_i1025"
 type="#_x0000_t75" alt="A screenshot of a computer&#10;&#10;Description automatically generated"
 style='width:452.5pt;height:203.5pt;visibility:visible;mso-wrap-style:square'>
 <v:imagedata src="file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_image007.png"
  o:title="A screenshot of a computer&#10;&#10;Description automatically generated"/>
</v:shape><![endif]--><!--[if !vml]-->![A screenshot of a computer

Description automatically generated](file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_image008.png)<!--[endif]-->

The [**Catalogue of Somatic Mutations in Cancer** **(COSMIC)**](https://cancer.sanger.ac.uk/cosmic/) database is another valuable resource that stores somatic mutation data and related information about human cancer . Since 2004, COSMIC has integrated coding mutations into its database, covering various genetic mechanisms through which somatic mutations contribute to cancer development. These mechanisms include non-coding mutations, gene fusions, copy-number variants, and drug resistance mutations. Additionally, the COSMIC website provides users with

the ability to visualize the 3D structure of proteins. In one of the research findings using COSMIC helped to the discovery that most melanoma cases contain specific mutations in the BRAF gene, which led to the development of BRAF inhibitors as an effective therapeutic option. However, these BRAF mutations are uncommon in hereditary melanomas, which account for only 10% of cases, and almost all BRAF mutations in melanoma are missense variations.

 

<!--[if gte vml 1]><v:shape
 id="Picture_x0020_3" o:spid="_x0000_s1026" type="#_x0000_t75" alt="A screenshot of a computer&#10;&#10;Description automatically generated"
 style='position:absolute;left:0;text-align:left;margin-left:0;margin-top:0;
 width:451.3pt;height:205.4pt;z-index:-251658240;visibility:visible;
 mso-wrap-style:square;mso-wrap-distance-left:9pt;mso-wrap-distance-top:0;
 mso-wrap-distance-right:9pt;mso-wrap-distance-bottom:0;
 mso-position-horizontal:absolute;mso-position-horizontal-relative:text;
 mso-position-vertical:absolute;mso-position-vertical-relative:text'>
 <v:imagedata src="file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_image009.png"
  o:title="A screenshot of a computer&#10;&#10;Description automatically generated"/>
 <w:wrap type="tight"/>
</v:shape><![endif]--><!--[if !vml]-->![A screenshot of a computer

Description automatically generated](file:///C:/Users/User/AppData/Local/Temp/msohtmlclip1/01/clip_image010.png)<!--[endif]-->
**References**

** **

Austin, B. K., Firooz, A., Valafar, H., & Blenda, A. V. (2023). An Updated Overview of Existing Cancer Databases and Identified Needs. _Biology_, _12_(8), 1152_These Tumor Databases You Have To Know_, 2 Sept. 2024, <https://www.cusabio.com/c-20916.html> <https://www.cancer.gov/ccg/research/genome-sequencing/tcga>

Kandoth, C., Schultz, N., Cherniack, A. D., Akbani, R., Liu, Y., Shen, H., ... & Levine, D. A. (2013). Integrated genomic characterization of endometrial carcinoma. _Nature, 497_(7447), 67-73.

Gao, J., Aksoy, B. A., Dogrusoz, U., Dresdner, G., Gross, B., Sumer, S. O., Sun, Y., Jacobsen, A., Sinha, R., Larsson, E., Cerami, E., Sander, C., & Schultz, N. (2013).Integrative analysis of complex cancer genomics and clinical profiles using the cBioPortal. _Science signaling_, _6_(269), pl1. https\://doi.org/10.1126/scisignal.2004088.

 Castellani G, Buccarelli M, Arasi MB, Rossi S, Pisanu ME, Bellenghi M, Lintas C, Tabolacci C. BRAF Mutations in Melanoma: Biological Aspects, Therapeutic Implications, and Circulating Biomarkers. Cancers (Basel). 2023 Aug 8;15(16):4026. doi: 10.3390/cancers15164026. PMID: 37627054; PMCID: PMC10452867.
