---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Ctrl+Shift+Enter*. 

```{r}
---
title: "R Notebook"
output: html_notebook
---

This is an [R Markdown](http://rmarkdown.rstudio.com) Notebook. When you execute code within the notebook, the results appear beneath the code. 

Try executing this chunk by clicking the *Run* button within the chunk or by placing your cursor inside it and pressing *Ctrl+Shift+Enter*. 

```{r}
#------------------------------------Downloading the dataset-----------------------------
#set your working directory
setwd("D:/HackBio/Intenship/stage 3/Results")

#load the necessary libraries
library("TCGAbiolinks")
library('ggplot2')
library(SummarizedExperiment)
library(data.table)
library(dplyr)
library(gplots)
library("biomaRt")

getProjectSummary("TCGA-SARC") #select your type
sb<- read.csv("Selected Barcodes.csv", row.names = 1)
#Preparing the query
SARCq<- GDCquery(project = "TCGA-SARC",                    #project
                 data.category = "Transcriptome Profiling", #to obtain RNA-seq data
                 data.type = "Gene Expression Quantification", #type of analysis
                 sample.type = c("Metastatic", "Primary Tumor", "Recurrent Tumor"), #to obtain only tumor samples and not normal samples
                 barcode = sb$x)
GDCdownload(SARCq) #downloading the dataset
sarc.data<-GDCprepare(SARCq) #Preparing the dataset
table(sarc.data$sample_type) #checking the type of samples to ensure no normal samples


#Obtaing the subgroups - age based classification

Metadata_1<- data.frame("barcode"= sarc.data$barcode,
                        "age_at_diagnosis"= sarc.data$age_at_diagnosis)

#checking for missing or 0 values in age
sum(is.na(Metadata_1$age_at_diagnosis)) #checking for na's
Metadata_na_omitted <-na.omit(Metadata_1) #removing NA
sum(is.na(Metadata_na_omitted$age_at_diagnosis))
sum(Metadata_na_omitted$age_at_diagnosis<0) #checking if there are any 0's

#since age_at_diagnosis is in days and conversion factor is 1 year = 365.25 days, we can use  14610 as the cutoff for 40 years and 6574.5 for 18 years
Metadata_na_omitted$age_group <- ifelse(Metadata_na_omitted$age_at_diagnosis<14610 & Metadata_na_omitted$age_at_diagnosis>=6574.5, "18-40", ">=40")
Group1_metadata<- Metadata_na_omitted[Metadata_na_omitted$age_group == '18-40',] #grouping samples with age between 18-40
Group2_metadata<- Metadata_na_omitted[Metadata_na_omitted$age_group == '>=40',] #grouping samples with age >=40

#select the unstranded dataset
sarc.raw.data<-assays(sarc.data) #using summarized experiments module 
dim(sarc.raw.data$unstranded) #since we are exploring both strands

#Downsizing to 20 samples in each age group (>=40 and 18-40)
selectedBarcodes<- c(sample(Group1_metadata$barcode,20), sample(Group2_metadata$barcode,20)) # selecting random samples from group 1 and group 2
selectedBarcodes
#obtaining the list of barcodes for future selection purpose
write.csv(selectedBarcodes, file="Selected Barcodes.csv", row.names= TRUE)


#obtaining the unstranded data of the selected barcodes
selectedData<-sarc.raw.data$unstranded[,c(selectedBarcodes)]
dim(selectedData)
View(selectedData)

#Obtaining the new Meta data file with only selected sample subset
new_meta_data<-(Metadata_na_omitted[Metadata_na_omitted$barcode %in% selectedBarcodes,])

#Downloading the and metadata file for ML analysis
#write.csv(new_meta_data, file ="Metadata_Final.csv", row.names =TRUE)

#Data normalization and filtering
#Normalization - based on sequencing depth and gene length to ensure comparable expression levels in the samples
normData<- TCGAanalyze_Normalization(tabDF = selectedData, geneInfo = geneInfoHT, method= "geneLength")
# Filtering out lowly expressed genes with a quantile normalization cutoff of 0.25
filtData<- TCGAanalyze_Filtering(tabDF = normData,
                                 method = "quantile",
                                 qnt.cut = 0.25) 

#Obtaining the filtered data for ML analysis
#write.csv(filtData, file ="Filtered_data_final.csv", row.names = TRUE)


```
```{r}
#------------------------------------------Diferential Gene Expression------------------------------------

#Differential Gene Expression analysis using TCGAanalyze_DEA
selectResults<-TCGAanalyze_DEA(mat1 = filtData[, c(selectedBarcodes)[1:20]], #1st group is 18-40
                               mat2 = filtData[, c(selectedBarcodes)[21:40]], #2nd group is >=40
                               Cond1type = "Age 18-40", #Defining the condition 1
                               Cond2type = "Age >=40", #Defining condition 2
                               pipeline = "edgeR") #using edgeR pipeline since it gives better fit to read counts data
View(selectResults) 

#Differential expression levels for the different conditions adds the average values for each group
selectResults.levels<-
  TCGAanalyze_LevelTab(selectResults,"Age 18-40" ,"Age >=40", 
                       filtData[,c(selectedBarcodes)[1:20]],
                       filtData[,c(selectedBarcodes)[21:40]])
dim(selectResults)
dim(selectResults.levels)

#Setting the logfc and p value filter
selectResults.levels$diff_exp <-"No"
selectResults.levels$diff_exp[selectResults.levels$logFC > 1.5 & selectResults.levels$FDR <0.005] <-"UP"
selectResults.levels$diff_exp[selectResults.levels$logFC < (-1.5) & selectResults.levels$FDR <0.005] <-"DOWN"
table(selectResults.levels$diff_exp)

#Generating a volcano plot
ggplot(data = selectResults.levels, aes(x = logFC, y = (- log10(FDR)), col = diff_exp)) +  #basic Volcano plot
  geom_vline(xintercept = c(-1.5, 1.5), col = "blue", linetype = 'dashed') + #checking the threshold to 1.5
  geom_hline(yintercept = -log10(0.005), col = "red", linetype = 'dashed') + #checking the significance to 0.005
  geom_point() +  #make a continuous plot
  scale_color_manual(values = c("blue", "grey", "red"), # to set the colors of our variable
                     labels = c("Downregulated", "Not significant", "Upregulated")) + # to set the labels we want to overwrite the categories from the dataframe (UP, DOWN, NO)
  labs(color = 'Gene condition', x= expression("log"[2]*"FoldChange"), y = expression("-log"[10]*"p-adj value"))+
  coord_cartesian(ylim = c(0, 40), xlim = c(-10, 10)) + # to set the limits of the axis
  ggtitle("Volcano plot of differentially expressed genes")

#Collecting the list of up and down regulated genes 
DE_results<- selectResults.levels[selectResults.levels$diff_exp == "UP" | selectResults.levels$diff_exp == "DOWN",]

#Upregulated and downregulated genes
upreg.genes<- rownames(subset(selectResults.levels[selectResults.levels$diff_exp =='UP',]))
dnreg.genes<- rownames(subset(selectResults.levels[selectResults.levels$diff_exp =='DOWN',]))

#Obtaining the list of significant differentially regulated genes
write.csv(DE_results, file ="DE_results.csv", row.names =TRUE)

#Top 5 Up and Down regulated gene list for literature study
mart<-useMart(biomart = "ensembl", dataset = "hsapiens_gene_ensembl")
up_top5<- getBM(attributes = c('ensembl_gene_id', 'hgnc_symbol'), #what are we converting
                filters = 'ensembl_gene_id', # data present in
                values = rownames(DE_results[order(DE_results$logFC, decreasing = TRUE),])[1:5],
                mart = mart)$hgnc_symbol
dn_top5<- getBM(attributes = c('ensembl_gene_id', 'hgnc_symbol'), 
                filters = 'ensembl_gene_id', 
                values = rownames(DE_results[order(DE_results$logFC, decreasing =TRUE),])[(nrow(DE_results)-4):nrow(DE_results)], 
                mart = mart)$hgnc_symbol

#------------------------------------------------Heatmap Visualization----------------------------------
dim(DE_results)
heat.data<-filtData[rownames(DE_results),] # selecting the genes that are significantly differentiated from the filtered data

#color based on the age groups of the samples -column colors
cancer.type<-c(rep("18-40", 20), rep(">=40",20))
ccodes<-c()
for(i in cancer.type)
{
  if(i == ">=40")
    ccodes <- c(ccodes,"red")
  else
    ccodes <- c(ccodes, "blue")
}
ccodes

#Plotting Heatmap
par(oma = c(1,1,1,1)) #Setting outter margins
par(mar = c(1,1,1,1)) #setting inner plot margins
par(cex.main = 0.75) #size of the title
heatmap.2(as.matrix(heat.data),
          col = hcl.colors(100, palette = "Blue-Red 3"), # Diverging palette
          Colv = F,                         # Cluster columns
          Rowv = F,                         # Cluster rows
          dendrogram = "none",              # No cluster both rows and columns
          trace = "none",                   # Remove trace lines
          scale = "row",                    # Standardizes rows (genes) across samples
          sepcolor = "black",               #separate the columns
          key = TRUE,                       # Show color key
          cexRow = 0.5,                     # Adjust row label size
          cexCol = 0.5,                     # Adjust column label size
          margins = c(9, 7),                # Adjust margins to fit labels
          main = "Heatmap", #Title
          xlab = "Samples",                 #X axis label
          ylab = "Genes",                   #Y axis label
          key.title = "Expression Level",
          ColSideColors = ccodes)  # colums are the samples and are color coded based on the previous for loop.
legend("topright", legend = c(">=40 age group", "18-40 age group"), fill = c("red", "blue"), title = "Column Colors", cex = 0.8)
```

```{r}
#---------------------------------Functional enrichment analysis using EA_plot-------------------------------------
#For the TCGA EA analysis tool the gene names have to be in genes names not Ensembl ids so convert using biomart

upreg.genes_symbol<- getBM(attributes = c('ensembl_gene_id', 'hgnc_symbol'), #what are we converting
                    filters = 'ensembl_gene_id', # data present in
                    values = upreg.genes, #the list
                    mart = mart)$hgnc_symbol # will have some empty genes names as not found in database
dnreg.genes_symbol<- getBM(attributes = c('ensembl_gene_id', 'hgnc_symbol'),
                    filters = 'ensembl_gene_id',
                    values = dnreg.genes,
                    mart = mart)$hgnc_symbol

#Performing Enrichment anlysis
up.EA<- TCGAanalyze_EAcomplete(TFname = "Upregulated", upreg.genes_symbol) # produces result based on BP, CC, MF and Pathways(P)
dn.EA<- TCGAanalyze_EAcomplete(TFname = "Downregulated", dnreg.genes_symbol)

#Visualization
TCGAvisualize_EAbarplot(tf = rownames(up.EA$ResBP),#Rownames
                        GOBPTab = up.EA$ResBP, #results for BP
                        GOMFTab = up.EA$ResMF, #results for MF
                        GOCCTab = up.EA$ResCC, #results for CC
                        PathTab = up.EA$ResPat, #results for PAthway
                        nRGTab = upreg.genes_symbol, #number of genes in the list
                        nBar = 10, #max number of bars is 5 but can be increased to 10
                        text.size = 2, # 2 
                        fig.width = 30, # size of figure
                        fig.height = 15) #generates a pdf in the working directory


TCGAvisualize_EAbarplot(tf = rownames(dn.EA$ResBP),
                        GOBPTab = dn.EA$ResBP, 
                        GOMFTab = dn.EA$ResMF, 
                        GOCCTab = dn.EA$ResCC, 
                        PathTab = dn.EA$ResPat, 
                        nRGTab = dnreg.genes_symbol, 
                        nBar = 10, 
                        text.size = 2, 
                        fig.width = 30,
                        fig.height = 15)

```
